### ISSUE TITLE
latest.yml hash for .exe release does not match

### ISSUE THREAD

**Issue by v4k0nd** [szTheory/exifcleaner#119]
I noticed that hash does not match for the .exe file. I few days back I noticed another project had the same issue, and they generated the latest.yaml files via Electron Builder. Could you check if it's just on my machine or does it really not match.
I checked two ways on my win machine:

`powershell> Get-FileHash .\ExifCleaner-Setup-3.4.0.exe -Algorithm SHA512`

`cmd> certutil -hashfile ExifCleaner-Setup-3.4.0.exe SHA512`

EDIT: accidentally pasted the commands for the other project that had this problem

**Comment by szTheory** [comment]
Thanks for the report! I just ran the sha 512 checksum for the Mac .dmg and it didn't match the release either. 

```bash
$ shasum -a 512 ExifCleaner-3.4.0.dmg 
d48c15121c24100a4b63dcadfc32b1b9ba38ddf0a0c5e095550a261e0372ef78f655ff6cae1e772bf32d5c68a0cfa44ce831ad1cb5bce7c0e9c9cef861ad89fb
```

Whereas `latest-mac.yml` from the releases page has the following

```yaml
version: 3.4.0
files:
  - url: ExifCleaner-3.4.0-mac.zip
    sha512: vCV4kCC9Wa/R+Veq/I9reo2IK1zZ92zvVKzteXOlj+sClxg9DHtj3E8y873pHWGpgC4xmhVfEjfQD5CYtAtdmg==
    size: 77893769
    blockMapSize: 82892
  - url: ExifCleaner-3.4.0.dmg
    sha512: 1IwVEhwkEApLY9yt/DKxubo43fCgxeCVVQomHgNy73j2Vf9srh53K/MtXGigz6RM6DGtHLW858Dpyc74Ya2J+w==
    size: 80393100
path: ExifCleaner-3.4.0-mac.zip
sha512: vCV4kCC9Wa/R+Veq/I9reo2IK1zZ92zvVKzteXOlj+sClxg9DHtj3E8y873pHWGpgC4xmhVfEjfQD5CYtAtdmg==
releaseDate: '2020-10-19T00:09:01.169Z'
```

I'm not sure what the issue is. Could be Electron Builder like you mentioned. I don't have a CI process in place I just run the cross-platform build on a Mac with the `yarn run publish` command from `package.json` which is an alias for this `yarn run compile && electron-builder --macos --linux --windows -p always` which compiles the webpack then builds the cross-platform targets and publishes to Github. 

I don't have a lot of time right now to dig into Electron Build problems but if someone can reproduce and open up an issue on the [electron-builder Github](https://github.com/electron-userland/electron-builder) or find others who are experiencing the same issues and report back that will really help. In the meantime if possible you could run ExifCleaner locally by [following the instructions here](https://github.com/sztheory/exifcleaner#run-the-app-in-dev-mode) for peace of mind.

Meanwhile I'm going to pin this issue to raise awareness and hopefully work toward a resolution using the wisdom of the crowd.

**Comment by v4k0nd** [comment]
I think I found something, there is a nearly two year old issue/answer in [electron-builder](https://github.com/electron-userland/electron-builder/issues/3913#issuecomment-504698845).
Under the mentioned issues, I found [this](https://github.com/electron-userland/electron-builder/issues/3119#issuecomment-405097164), where it says that electron-builder wont just calculate the SHA512 and output it in ascii, but after calculating the SHA512, it takes the binary and converts it to base64. I should have noticed that it is not just normal ascii since because the sha512 has a `==` at the end, which indicates padding for the encoding, usually.

However this does not answer me how to check it "manually" (only cmd / powershell / bash commands). 

I did find something for bash in [this issue](https://github.com/th-ch/youtube-music/issues/69#issuecomment-735396905) but it did not calculate the correct hash, nor any combination from the mentioned stackexchange link.

If I manage to find an answer I will try to keep this issue updated.

**Comment by szTheory** [comment]
I just released ExifCleaner 3.5.0, and was able to verify the checksum of every Linux, Windows, and Mac release with the following command*, which I took from the link you provided.

```bash
sha512sum ExifCleaner-Setup-3.5.0.exe | cut -f1 -d\ | xxd -r -p | base64
```

*Except for Linux RPM which for some reason did not generate a checksum in latest-linux.yml. Must be an electron-builder issue, will have to look into this later.

Thanks again for your help on this. I've also added instructions for this to the README. What do you think, does that satisfy the issue at hand? I'll close this issue for now but if there is anything that could be done to make the process better, I can reopen it.

**Comment by kkang-ss** [comment]
**Windows**: In Powershell, this will work for you
```
$hex = (certutil -hashfile <FullFilePath> sha512 | Select -Index 1) -replace '..', '0x$& '
[Convert]::ToBase64String([byte[]] -split $hex)
```
